blocks/form/form.js

```
@@ -1,36 +1,91 @@
+    head.append(script);
+    return script;
+  }
+  return script;
+}
+
 function constructPayload(form) {
   const payload = {};
+  const attachments = {};
   [...form.elements].forEach((fe) => {
     if (fe.type === 'checkbox') {
       if (fe.checked) payload[fe.id] = fe.value;
+    } else if (fe.type === 'file' && fe.files?.length > 0) {
+      attachments[fe.name] = fe.files;
     } else if (fe.id) {
       payload[fe.id] = fe.value;
     }
   });
-  return payload;
+  return { payload, attachments };
 }
 
-async function submitForm(form) {
-  const payload = constructPayload(form);
-  const resp = await fetch(form.dataset.action, {
+async function submissionFailure(error, form) {
+  alert(error); // TODO define error mechansim
+  form.setAttribute('data-submitting', 'false');
+  form.querySelector('button[type="submit"]').disabled = false;
+}
+
+async function submitForm(form, token) {
+  const url = `${FORM_SUBMIT_ENDPOINT}${form.dataset.action}`;
+  const { payload, attachments } = constructPayload(form);
+  let headers = {
+    'Content-Type': 'application/json',
+  };
+  let body = JSON.stringify({ data: payload, token });
+  if (attachments && Object.keys(attachments).length > 0) {
+    headers = {};
+    body = new FormData();
+    const fileNames = [];
+    Object.entries(attachments).forEach(([dataRef, files]) => {
+      fileNames.push(dataRef);
+      [...files].forEach((file) => body.append(dataRef, file));
+    });
+    body.append('token', token);
+    body.append('fileFields', JSON.stringify(fileNames));
+    body.append('data', JSON.stringify(payload));
+  }
+  fetch(url, {
     method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify({ data: payload }),
+    headers,
+    body,
+  }).then(async (response) => {
+    if (response.ok) {
+      sampleRUM('form:submit');
+      window.location.href = form.dataset?.redirect || 'thankyou';
+    } else {
+      const error = await response.text();
```

API won't reject in case of HTTP 4xx or 5xx error codes. You need to handle this manually. You already handle non-200 status codes which is good but it's better to throw an error after response.text() so it can be caught by the catch block.

```
    if (!response.ok) {
      const error = await response.text();
      throw new Error(error);
    }
```